gear/gear/auth.py

```diff
@@ -53,48 +53,41 @@ def __init__(self):
             self._load_userdata, TEN_SECONDS_IN_NANOSECONDS, 100, 'session_userdata_cache'
         )
 
-    def rest_authenticated_users_only(self, fun: AuthenticatedAIOHTTPHandler) -> AIOHTTPHandler:
-        async def wrapped(request: web.Request) -> web.StreamResponse:
-            userdata = await self._userdata_from_rest_request(request)
-            if not userdata:
-                web_userdata = await self._userdata_from_web_request(request)
-                if web_userdata:
-                    return web.HTTPUnauthorized(reason="provided web auth to REST endpoint")
-                raise web.HTTPUnauthorized()
-            return await fun(request, userdata)
-
-        return wrapped
-
-    def web_authenticated_users_only(
-        self, redirect: bool = True
+    def authenticated_users_only(
+        self, redirect: Optional[bool] = None
     ) -> Callable[[AuthenticatedAIOHTTPHandler], AIOHTTPHandler]:
         def wrap(fun: AuthenticatedAIOHTTPHandler):
             @wraps(fun)
             async def wrapped(request: web.Request) -> web.StreamResponse:
-                userdata = await self._userdata_from_web_request(request)
+                userdata = await self._userdata_from_request(request)
                 if not userdata:
-                    rest_userdata = await self._userdata_from_rest_request(request)
-                    if rest_userdata:
-                        raise web.HTTPUnauthorized(reason="provided REST auth to web endpoint")
-                    _web_unauthenticated(request, redirect)
+                    if redirect is None:
+                        # Only web routes should redirect by default
+                        should_redirect = '/api/' not in request.path
```

Comment:
it should start with `/api/`.

Reply:
Not exactly the case, it could also start with `/dgoldste/batch/api/`, but I see your point on looking at prefixes.